diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0507e30..2e835eb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -88,7 +88,7 @@ set(CMAKE_VISIBILITY_INLINES_HIDDEN ON)
 add_compile_options("-fmacro-prefix-map=${PROJECT_SOURCE_DIR}/=")
 
 # warnings
-add_compile_options(-Wall -Wextra -Werror)
+add_compile_options(-Wall -Wextra)
 add_compile_options(-Wcast-align -Wformat-nonliteral -Wmissing-format-attribute -Wredundant-decls -Wsign-compare -Wsign-conversion -Wtype-limits -Wuninitialized -Wwrite-strings)
 add_compile_options(-Werror=unused-result -Wodr)
 
