diff --git a/dnf5-plugins/reposync_plugin/CMakeLists.txt b/dnf5-plugins/reposync_plugin/CMakeLists.txt
index 7e3f7c1..6e3ac31 100644
--- a/dnf5-plugins/reposync_plugin/CMakeLists.txt
+++ b/dnf5-plugins/reposync_plugin/CMakeLists.txt
@@ -8,7 +8,7 @@ add_library(reposync_cmd_plugin MODULE reposync.cpp reposync_cmd_plugin.cpp)
 set_target_properties(reposync_cmd_plugin PROPERTIES PREFIX "")
 
 target_link_libraries(reposync_cmd_plugin PRIVATE libdnf5 libdnf5-cli)
-target_link_libraries(reposync_cmd_plugin PRIVATE dnf5)
+target_link_libraries(reposync_cmd_plugin PRIVATE dnf5_core)
 
 install(TARGETS reposync_cmd_plugin LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/dnf5/plugins/)
 install(DIRECTORY "config/usr/" DESTINATION "${CMAKE_INSTALL_PREFIX}")
