diff --git a/dnf5-plugins/config-manager_plugin/CMakeLists.txt b/dnf5-plugins/config-manager_plugin/CMakeLists.txt
index da04603..3785642 100644
--- a/dnf5-plugins/config-manager_plugin/CMakeLists.txt
+++ b/dnf5-plugins/config-manager_plugin/CMakeLists.txt
@@ -18,7 +18,7 @@ find_package(ZLIB REQUIRED)
 
 target_link_libraries(config-manager_cmd_plugin PRIVATE ZLIB::ZLIB)
 target_link_libraries(config-manager_cmd_plugin PRIVATE libdnf5 libdnf5-cli)
-target_link_libraries(config-manager_cmd_plugin PRIVATE dnf5)
+target_link_libraries(config-manager_cmd_plugin PRIVATE dnf5_core)
 
 install(TARGETS config-manager_cmd_plugin LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/dnf5/plugins/)
 
