diff --git a/dnf5/context.cpp b/dnf5/context.cpp
index 5a2cac3..bb76b54 100644
--- a/dnf5/context.cpp
+++ b/dnf5/context.cpp
@@ -52,6 +52,8 @@
 #include <tuple>
 #include <utility>
 
+#include <zos-filesystem.h>
+
 namespace fs = std::filesystem;
 
 namespace dnf5 {
@@ -455,8 +457,8 @@ void Context::Impl::download_and_run(libdnf5::base::Transaction & transaction) {
                 _("Location \"{}\" already contains a stored transaction, it will be overwritten."),
                 transaction_store_path.string()));
             if (libdnf5::cli::utils::userconfirm::userconfirm(config)) {
-                std::filesystem::remove_all(packages_location);
-                std::filesystem::remove_all(comps_location);
+                portable_remove_all(packages_location);
+                portable_remove_all(comps_location);
             } else {
                 throw libdnf5::cli::AbortedByUserError();
             }
