diff --git a/dnf5-plugins/builddep_plugin/CMakeLists.txt b/dnf5-plugins/builddep_plugin/CMakeLists.txt
index b8640e1..143e668 100644
--- a/dnf5-plugins/builddep_plugin/CMakeLists.txt
+++ b/dnf5-plugins/builddep_plugin/CMakeLists.txt
@@ -7,7 +7,7 @@ add_library(builddep_cmd_plugin MODULE builddep.cpp builddep_cmd_plugin.cpp)
 # disable the 'lib' prefix in order to create builddep_cmd_plugin.so
 set_target_properties(builddep_cmd_plugin PROPERTIES PREFIX "")
 
-find_library(RPMBUILD NAMES rpmbuild REQUIRED)
+find_library(RPMBUILD NAMES librpmbuild REQUIRED)
 target_link_libraries(builddep_cmd_plugin PRIVATE ${RPMBUILD})
 
 pkg_check_modules(RPM REQUIRED rpm)
