diff --git a/dnf5-plugins/needs_restarting_plugin/CMakeLists.txt b/dnf5-plugins/needs_restarting_plugin/CMakeLists.txt
index cf29014..3caafda 100644
--- a/dnf5-plugins/needs_restarting_plugin/CMakeLists.txt
+++ b/dnf5-plugins/needs_restarting_plugin/CMakeLists.txt
@@ -9,7 +9,7 @@ set_target_properties(needs_restarting_cmd_plugin PROPERTIES PREFIX "")
 
 include(sdbus_cpp)
 
-target_link_libraries(needs_restarting_cmd_plugin PRIVATE dnf5 libdnf5 libdnf5-cli ${SDBUS_CPP_LIBRARIES})
+target_link_libraries(needs_restarting_cmd_plugin PRIVATE dnf5_core libdnf5 libdnf5-cli ${SDBUS_CPP_LIBRARIES})
 
 install(TARGETS needs_restarting_cmd_plugin LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/dnf5/plugins/)
 
