diff --git a/include/libdnf5/base/base.hpp b/include/libdnf5/base/base.hpp
index b0084f3..756d789 100644
--- a/include/libdnf5/base/base.hpp
+++ b/include/libdnf5/base/base.hpp
@@ -40,6 +40,10 @@
 
 #include <libdnf5/utils/locker.hpp>
 
+#ifdef __MVS__
+FILE *memstream_open(char **out_buf, size_t *out_len);
+int memstream_finish(FILE *fp, char **out_buf, size_t *out_len);
+#endif
 
 namespace libdnf5::module {
 
