diff --git a/libdnf5-cli/CMakeLists.txt b/libdnf5-cli/CMakeLists.txt
index db1a6fcc..469a042c 100644
--- a/libdnf5-cli/CMakeLists.txt
+++ b/libdnf5-cli/CMakeLists.txt
@@ -28,7 +28,7 @@ list(APPEND LIBDNF5_CLI_PC_REQUIRES_PRIVATE)
 include_directories(.)
 
 # build libdnf5-cli.so
-add_library(libdnf5-cli SHARED ${LIBDNF5_CLI_SOURCES})
+add_library(libdnf5-cli STATIC ${LIBDNF5_CLI_SOURCES})
 set(DNF_CLI_SO_VERSION 3)
 set_target_properties(libdnf5-cli PROPERTIES OUTPUT_NAME "dnf5-cli")
 set_target_properties(libdnf5-cli PROPERTIES SOVERSION ${DNF_CLI_SO_VERSION})
@@ -46,9 +46,11 @@ target_link_libraries(libdnf5-cli PRIVATE common_obj)
 
 target_link_libraries(libdnf5-cli PUBLIC libdnf5)
 
-pkg_check_modules(LIBFMT REQUIRED fmt)
-list(APPEND LIBDNF5_CLI_PC_REQUIRES "${LIBFMT_MODULE_NAME}")
-target_link_libraries(libdnf5-cli PUBLIC ${LIBFMT_LIBRARIES})
+#pkg_check_modules(LIBFMT REQUIRED fmt)
+#list(APPEND LIBDNF5_CLI_PC_REQUIRES "${LIBFMT_MODULE_NAME}")
+#target_link_libraries(libdnf5-cli PUBLIC ${LIBFMT_LIBRARIES})
+find_package(fmt REQUIRED)
+target_link_libraries(libdnf5-cli PUBLIC fmt::fmt-header-only)
 
 pkg_check_modules(SMARTCOLS REQUIRED smartcols)
 list(APPEND LIBDNF5_CLI_PC_REQUIRES_PRIVATE "${SMARTCOLS_MODULE_NAME}")
