diff --git a/libdnf5-cli/progressbar/progress_bar.cpp b/libdnf5-cli/progressbar/progress_bar.cpp
index 9098296..214982d 100644
--- a/libdnf5-cli/progressbar/progress_bar.cpp
+++ b/libdnf5-cli/progressbar/progress_bar.cpp
@@ -103,7 +103,8 @@ const ProgressBar::Impl::MessageMetrics & ProgressBar::Impl::get_message_metrics
         if (bytes_consumed == 0 || bytes_consumed >= SIZE_MAX - 1) {
             break;
         }
-        auto char_width = static_cast<std::size_t>(wcwidth(wc));
+        int width = wcwidth(wc);
+        auto char_width = width >= 0 ? static_cast<std::size_t>(width) : 0;
 
         // If the character doesn't fit, wrap to the next line
         if (current_column + char_width > terminal_width) {
@@ -308,7 +309,7 @@ void ProgressBar::update() {
             static_cast<int32_t>(static_cast<float>(p_impl->ticks) / static_cast<float>(p_impl->total_ticks) * 100);
     }
 
-    auto now = std::chrono::high_resolution_clock::now();
+    auto now = std::chrono::system_clock::now();
 
     // compute the current speed (ticks per second)
     auto delta = now - p_impl->current_speed_window_start;
