diff --git a/test/libdnf5/conf/test_conf.cpp b/test/libdnf5/conf/test_conf.cpp
index a031481..c91542e 100644
--- a/test/libdnf5/conf/test_conf.cpp
+++ b/test/libdnf5/conf/test_conf.cpp
@@ -17,8 +17,9 @@
 // You should have received a copy of the GNU General Public License
 // along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
-
+#define system system_dont_use
 #include "test_conf.hpp"
+#undef system
 
 #include "../shared/utils.hpp"
 
diff --git a/test/libdnf5/conf/test_option.cpp b/test/libdnf5/conf/test_option.cpp
index bca457e..077bdd6 100644
--- a/test/libdnf5/conf/test_option.cpp
+++ b/test/libdnf5/conf/test_option.cpp
@@ -17,8 +17,9 @@
 // You should have received a copy of the GNU General Public License
 // along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
-
+#define system system_dont_use
 #include "test_option.hpp"
+#undef system
 
 #include "../shared/utils.hpp"
 
