diff --git a/test/libdnf5/system/test_state.cpp b/test/libdnf5/system/test_state.cpp
index cdc268d..16e8558 100644
--- a/test/libdnf5/system/test_state.cpp
+++ b/test/libdnf5/system/test_state.cpp
@@ -18,11 +18,13 @@
 // along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 
-#include "test_state.hpp"
+#define system system_dont_use
+#include "libdnf5/utils/fs/file.hpp"
+#undef system
 
 #include "../shared/utils.hpp"
 
-#include "libdnf5/utils/fs/file.hpp"
+#include "test_state.hpp"
 
 
 CPPUNIT_TEST_SUITE_REGISTRATION(StateTest);
