diff --git a/test/dnf5-plugins/automatic_plugin/EmailMessage.cpp b/test/dnf5-plugins/automatic_plugin/EmailMessage.cpp
index 565ed48..3bfe2ed 100644
--- a/test/dnf5-plugins/automatic_plugin/EmailMessage.cpp
+++ b/test/dnf5-plugins/automatic_plugin/EmailMessage.cpp
@@ -29,9 +29,12 @@ using namespace std;
 // Redefine std::chrono::system_clock::now() used by dnf5::EmailMessage::str()
 // to always return fixed time 1970-01-01T00:00:00.000000.
 std::chrono::time_point<std::chrono::system_clock> std::chrono::system_clock::now() noexcept {
+    setenv("TZ", "UTC", 1);
+    tzset();
     return std::chrono::time_point<std::chrono::system_clock>();
 }
 
+
 void EmailMessageTest::test_email_emitter() {
     std::stringstream body{"text"};
     dnf5::EmailMessage message;
