diff --git a/test/libdnf5-cli/run_tests.cpp b/test/libdnf5-cli/run_tests.cpp
index 78242a1..ddcfbb6 100644
--- a/test/libdnf5-cli/run_tests.cpp
+++ b/test/libdnf5-cli/run_tests.cpp
@@ -31,16 +31,16 @@
 
 class TimingListener : public CppUnit::TestListener {
 public:
-    void startTest(CppUnit::Test *) override { start = std::chrono::high_resolution_clock::now(); }
+    void startTest(CppUnit::Test *) override { start = std::chrono::system_clock::now(); }
 
     void endTest(CppUnit::Test *) override {
-        auto end = std::chrono::high_resolution_clock::now();
+        auto end = std::chrono::system_clock::now();
         auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
         std::cout << " (duration: " << duration << "ms)";
     }
 
 private:
-    std::chrono::high_resolution_clock::time_point start = std::chrono::high_resolution_clock::from_time_t(0);
+    std::chrono::system_clock::time_point start = std::chrono::system_clock::from_time_t(0);
 };
 
 
diff --git a/test/libdnf5-cli/test_progressbar.cpp b/test/libdnf5-cli/test_progressbar.cpp
index 64442b6..de5e2a7 100644
--- a/test/libdnf5-cli/test_progressbar.cpp
+++ b/test/libdnf5-cli/test_progressbar.cpp
@@ -55,7 +55,11 @@ void ProgressbarTest::test_progress_bar_multi_byte_character() {
         "Created symlink '/etc/systemd/user/sockets.target.wants/pipewire.socket' \342\206\222 "
         "'/usr/lib/systemd/user/pipewire.socket'.");
     auto num_lines = progress_bar->calculate_messages_terminal_lines(106);
+#ifdef __MVS__
+    std::size_t expected = 2U;
+#else
     std::size_t expected = strcmp(setlocale(LC_ALL, NULL), "C") == 0 ? 1U : 2U;
+#endif
     CPPUNIT_ASSERT_EQUAL(expected, num_lines);
 }
 
